### PR TITLE
Temporary to enable changes in the constraints within matrix-free

### PR DESCRIPTION
**Pull request by mschreter**
This PR introduces a temporary work-around to rebuild matrix-free if the constraint matrix has changed. This occurs in the melt pool simulations, where the solid domain is considered by means of setting the constraints for the velocity field to zero. I've moved the test case from tests_not_robust to tests since at least the test should now be fine. Let's see what CI says ;-).

### CODE REVIEW

File: include/meltpooldg/flow/two_phase_flow_problem.hpp [hunk 1]
```diff
@@ -55,6 +55,9 @@ namespace MeltPoolDG::Flow
 
       while (!time_iterator.is_finished())
         {
+          if (base_in->parameters.mp.set_velocity_to_zero_in_solid)
+            scratch_data->build();
```
Comment (peterrum) [inline include/meltpooldg/flow/two_phase_flow_problem.hpp:59]: Could you add a comment here so that we remember in a couple of weeks what we did here/

Comment (mschreter) [reply include/meltpooldg/flow/two_phase_flow_problem.hpp:68]: Yes that's a good point.

**Comment by peterrum** [comment]
```
98% tests passed, 1 tests failed out of 44

Total Test time (real) = 1880.07 sec

The following tests FAILED:
	 34 - tests/thermo_capillary_two_droplets.mpirun=4.debug (Failed)
```

**Comment by mschreter** [comment]
Mmh, that's weird. I do not get any problems locally. The stacktrace of CI says:

@nmuch: Did you observe similar problems?
```
34: --------------------------------------------------------
34: 
34: --------------------------------------------------------
34: An error occurred in line <899> of file </usr/local/include/deal.II/dofs/dof_accessor.templates.h> in function
34:     static void dealii::internal::DoFAccessorImplementation::Implementation::process_dof_indices(const dealii::DoFAccessor<structdim, dim, spacedim, level_dof_access>&, DoFIndicesType&, unsigned int, const DoFOperation&, bool) [with int dim = 2; int spacedim = 2; bool level_dof_access = false; int structdim = 1; DoFIndicesType = std::vector<unsigned int>; DoFOperation = dealii::internal::DoFAccessorImplementation::Implementation::DoFIndexGetter<2, 2, false, 1>]
34: The violated condition was: 
34:     static_cast<typename ::dealii::internal::argument_type<void( typename std::common_type<decltype(dof_indices.size()), decltype(n_dof_indices(accessor, fe_index, count_level_dofs))>::type)>::type>(dof_indices.size()) == static_cast<typename ::dealii::internal::argument_type<void( typename std::common_type<decltype(dof_indices.size()), decltype(n_dof_indices(accessor, fe_index, count_level_dofs))>::type)>::type>(n_dof_indices(accessor, fe_index, count_level_dofs))
34: Additional information: 
34:     Dimension 6 not equal to 0.
34: 
34: Stacktrace:
34: -----------
34: #0  /home/dealii/adaflo-build/libadaflo.so: void dealii::internal::DoFAccessorImplementation::Implementation::process_dof_indices<2, 2, false, 1, std::vector<unsigned int, std::allocator<unsigned int> >, dealii::internal::DoFAccessorImplementation::Implementation::DoFIndexGetter<2, 2, false, 1> >(dealii::DoFAccessor<1, 2, 2, false> const&, std::vector<unsigned int, std::allocator<unsigned int> >&, unsigned int, dealii::internal::DoFAccessorImplementation::Implementation::DoFIndexGetter<2, 2, false, 1> const&, bool)
34: #1  /home/dealii/adaflo-build/libadaflo.so: void dealii::internal::DoFAccessorImplementation::Implementation::get_dof_indices<2, 2, false, 1>(dealii::DoFAccessor<1, 2, 2, false> const&, std::vector<unsigned int, std::allocator<unsigned int> >&, unsigned int)
34: #2  /home/dealii/adaflo-build/libadaflo.so: dealii::DoFAccessor<1, 2, 2, false>::get_dof_indices(std::vector<unsigned int, std::allocator<unsigned int> >&, unsigned int) const
34: #3  /usr/local/lib/libdeal_II.g.so.9.3.0-pre: 
34: #4  /usr/local/lib/libdeal_II.g.so.9.3.0-pre: void dealii::DoFTools::make_periodicity_constraints<dealii::TriaIterator<dealii::DoFAccessor<1, 2, 2, false> >, double>(dealii::TriaIterator<dealii::DoFAccessor<1, 2, 2, false> > const&, dealii::identity<dealii::TriaIterator<dealii::DoFAccessor<1, 2, 2, false> > >::type const&, dealii::AffineConstraints<double>&, dealii::ComponentMask const&, bool, bool, bool, dealii::FullMatrix<double> const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, double)
34: #5  /usr/local/lib/libdeal_II.g.so.9.3.0-pre: void dealii::DoFTools::make_periodicity_constraints<2, 2, double>(std::vector<dealii::GridTools::PeriodicFacePair<dealii::DoFHandler<2, 2>::cell_iterator>, std::allocator<dealii::GridTools::PeriodicFacePair<dealii::DoFHandler<2, 2>::cell_iterator> > > const&, dealii::AffineConstraints<double>&, dealii::ComponentMask const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, double)
34: #6  /usr/local/lib/libdeal_II.g.so.9.3.0-pre: void dealii::DoFTools::make_periodicity_constraints<2, 2, double>(dealii::DoFHandler<2, 2> const&, unsigned int, unsigned int, unsigned int, dealii::AffineConstraints<double>&, dealii::ComponentMask const&, double)
34: #7  /home/dealii/adaflo-build/libadaflo.so: NavierStokes<2>::initialize_data_structures()
34: #8  /home/dealii/adaflo-build/MeltPoolDG-build/meltpooldg: MeltPoolDG::Flow::AdafloWrapper<2>::reinit_1()
34: #9  /home/dealii/adaflo-build/MeltPoolDG-build/meltpooldg: MeltPoolDG::Flow::TwoPhaseFlowProblem<2>::setup_dof_system(std::shared_ptr<MeltPoolDG::SimulationBase<2, 2> >, bool)
34: #10  /home/dealii/adaflo-build/MeltPoolDG-build/meltpooldg: MeltPoolDG::Flow::TwoPhaseFlowProblem<2>::initialize(std::shared_ptr<MeltPoolDG::SimulationBase<2, 2> >)
34: #11  /home/dealii/adaflo-build/MeltPoolDG-build/meltpooldg: MeltPoolDG::Flow::TwoPhaseFlowProblem<2>::run(std::shared_ptr<MeltPoolDG::SimulationBase<2, 2> >)
34: #12  /home/dealii/adaflo-build/MeltPoolDG-build/meltpooldg: void MeltPoolDG::Simulation::run_simulation<double>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, ompi_communicator_t*)
34: #13  /home/dealii/adaflo-build/MeltPoolDG-build/meltpooldg: main
34: --------------------------------------------------------
34: 
34: Calling MPI_Abort now.
34: To break execution in a GDB session, execute 'break MPI_Abort' before running. You can also put the following into your ~/.gdbinit:
34:   set breakpoint pending on
34:   break MPI_Abort
34:   set breakpoint pending auto
34: 
34: --------------------------------------------------------------------------
34: MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
34: with errorcode 255.
34: 
34: NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
34: You may or may not see output from other processes, depending on
34: exactly when Open MPI kills them.
34: --------------------------------------------------------------------------
34: Calling MPI_Abort now.
34: To break execution in a GDB session, execute 'break MPI_Abort' before running. You can also put the following into your ~/.gdbinit:
34:   set breakpoint pending on
34:   break MPI_Abort
34:   set breakpoint pending auto
34: [3cf26794d2f6:02709] 2 more processes have sent help message help-mpi-api.txt / mpi-abort
34: [3cf26794d2f6:02709] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
```

**Comment by peterrum** [comment]
> Mmh, that's weird. I do not get any problems locally. 

Should be fixed by https://github.com/dealii/dealii/pull/12040.

**Review by peterrum — APPROVED** [review]

**Pull request merged**